### PR TITLE
pgraph.c: refactor clip dimensions

### PR DESCRIPTION
**Pull request by MasonT8198**
After experimenting with a few different solutions for issue #375 this solution (partially suggested by xbox7887) corrects the malloc(); corrupted top size issue seen in Digimon World 4/X, fixing #375

### CODE REVIEW

**Comment by MasonT8198** [comment]
Also Seems to fix Predator: Concrete Jungle

File: hw/xbox/nv2a/pgraph.c [hunk 1]
```diff
@@ -5319,8 +5319,8 @@ static void pgraph_populate_surface_binding_entry(NV2AState *d, bool color,
         /* Since we determine surface dimensions based on the clipping
          * rectangle, make sure to include the surface offset as well.
          */
-        width += pg->surface_shape.clip_x;
-        height += pg->surface_shape.clip_y;
+        width -= width % ( 1 << pg->surface_shape.clip_x);
+        height -= height % ( 1 << pg->surface_shape.clip_y);
```
Comment (mzji) [inline hw/xbox/nv2a/pgraph.c]:
```suggestion
        width &= ~((1 << pg->surface_shape.clip_x) - 1);
        height &= ~((1 << pg->surface_shape.clip_y) - 1);
```

Comment (MasonT8198) [reply hw/xbox/nv2a/pgraph.c]: Built and tested, also works as well implementation would most likely be up to @mborgerson

**Comment by mborgerson** [comment]
The logic here is not right, but I've fixed it elsewhere and will merge the fix soon. I'll close this now. Thank you for the PR!

**Pull request closed without merging**